### DISCUSSION TITLE
How can I send this message?

### DISCUSSION THREAD

**Discussion by gegne** [Rapptz/discord.py#6387] [Q&A] [answered]
Here is my code:

```
import discord
import random

import asyncio
import time
    
gep = ['a', 'b', 'c']

class MyClient(discord.Client):

    async def on_ready(self):
        print('Logged in as')
        print(self.user.name)
        print(self.user.id)
        print('------')
        while True:
            print(456)
            channel = client.get_channel("channel1")
            await channel.send('hello')
            print(789)
            time.sleep(900)
            
    async def on_message(self, message):
        if message.content.startswith('-random'):
           await message.channel.send(random.choice(gep))

client = MyClient()

client.run('token')
```

I replaced some sensitive information for this discussion, but I do have the 'token' field filled in. I have two questions: First, will this work if i put a string in the parentheses?

> channel` = client.get_channel("channel1")

Or does it need to be some sort of ID? Second, he main problem is I get this warning when I try to send the 'hello' message to a specific channel: 

> line 54, in on_ready await channel.send('hello')

> AttributeError: 'NoneType' object has no attribute 'send'

Can someone please educate me on what could cause that and how I can fix it? Thank you.

**Comment by bandwiches** [comment, marked as answer]
[Client.get_channel](https://discordpy.readthedocs.io/en/latest/api.html#discord.Client.get_channel) requires an `id` parameter which is not the same as the string label of the channel itself.  So you would actually be passing the `Discord ID` of the channel, rather than a string name of the channel.

So, this will not work:
> channel = client.get_channel("channel1")

This actually leads into your error.  In your specific case `channel = client.get_channel("channel1")` is the same as `channel = None`  because the call failed to look up a channel and thus defaults to `None`.  Next step is that you try `channel.send('hello')` and get the error because `channel = None` at the time of calling `send()`.

Hopefully this wasn't too confusing!

Edit:  if you pass an actual `Discord Channel ID` to `client.get_channel()` and it's able to retrieve that channel, then you won't see the error and all should work as far as I can see.

**Reply by gegne** [reply]
Thank you, I understand now.

**Comment by 15696** [comment]
Don't use `time.sleep`, it [blocks the asyncio event loop](https://discordpy.readthedocs.io/en/latest/faq.html#what-does-blocking-mean). Use `asyncio.sleep` instead.

**Reply by 15696** [reply]
Also, the [tasks extension](https://discordpy.readthedocs.io/en/latest/ext/tasks/) is probably what you're looking for. Then you can use something like:
```
@tasks.loop(seconds = 900)
async def loop():
            channel = client.get_channel(id)
            await channel.send('hello')
```
And use `loop.start()` in your on_ready.

**Reply by bandwiches** [reply]
Seconding the use of `asyncio.sleep` inside async event loops.
